packages/celer-web-app/src/core/context/SettingContext.ts

```diff
@@ -1,22 +1,41 @@
 import React, { useContext } from "react";
-import { SplitType } from "core/compiler";
-import { MapDisplayMode, SplitTypeSetting, Theme } from "core/settings";
+import { MapDisplayModes, Themes, defaultSplitSetting, getNextMap, getNextTheme } from "core/settings";
+import { LocalStorageWrapper } from "data/storage";
 import { Consumer, emptyObject } from "data/util";
 
-export interface AppSetting {
-    mapDisplayMode: MapDisplayMode,
-    theme: Theme,
-    splitSetting: SplitTypeSetting<boolean>,
-    enableSubsplits: boolean,
+export type Setting =
+{
+	theme: typeof Themes.Default
+	mapDisplay: typeof MapDisplayModes.Auto
+	splitSettings: typeof defaultSplitSetting
+	enableSubsplits: boolean
```

Comment:
Why are you deleting the existing type declaration and using `typeof`? Let's not overuse `typeof` to make the compiler happy. You won't need to change this code if you update the types `MapDisplayMode`, `Theme`, etc to your new system